### PR TITLE
Pubmed query not working

### LINKED ISSUE #8327: Pubmed query not working
**Describe the bug**
When a new submission is started from a query against Pubmed external system, the query does not return any result. 

**To Reproduce**
1. login to dspace7 and go to MyDSpace
2. start a new Publication submission from external source
3. enter a query string and select “Pubmed” as external source

Call to backend performing Pubmed query will fail with 500 error (https://api7.dspace.org/server/api/integration/externalsources/pubmed/entries?sort=id,ASC&page=0&size=10&query=covid19)

**Expected behavior**
A query returning results.

Running REST locally to investigate the bug, we found out that the first ever query on Pubmed performed when REST backend is up and running returns results, whereas next queries (second, third,...) ends up with no results and 500 error.

Our proposal is, instead of just fixing this bug, operate a refactoring to Pubmed integration and align it to the ones recently merged in DSpace (e.g. CrossRef), then add some tests on Pubmed LiveImport connector refactored. 
We (4Science) can take in charge this activity. Our estimation for refactoring and test addition is 8/12 hours of work

### PR DESCRIPTION
**Pull request by Micheleboychuk**
Fixes https://github.com/DSpace/DSpace/issues/8327

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_

- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & integration tests). Exceptions may be made if previously agreed upon.
- [x] My PR passes Checkstyle validation based on the [Code Style Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Style+Guide).
- [x] My PR includes Javadoc for _all new (or modified) public methods and classes_. It also includes Javadoc for large or complex private methods.
- [x] My PR passes all tests and includes new/updated Unit or Integration Tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new, third-party dependencies (in any `pom.xml`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [x] If my PR modifies the REST API, I've linked to the REST Contract page (or open PR) related to this change.

### CODE REVIEW

File: dspace/config/modules/external-providers.cfg [hunk 1]
```diff
@@ -25,4 +25,9 @@ vufind.url.search = https://vufind.org/advanced_demo/api/v1/search
 
 scielo.url = https://search.scielo.org/?output=ris&q=
 
+#################################################################
+#----------------------   Pubmed   -----------------------------#
+#---------------------------------------------------------------#
+pabmed.url.search = https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esearch.fcgi
+pabmed.url.fetch = https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi
```
Comment (tdonohue) [inline dspace/config/modules/external-providers.cfg:32]: Please fix the misspelling of Pubmed in these configs.  Should be `pubmed.url.*` obviously

File: dspace-api/src/main/java/org/dspace/importer/external/pubmed/service/PubmedImportMetadataSourceServiceImpl.java [hunk 2]
```diff
@@ -396,40 +407,57 @@ public FindMatchingRecords(Query q) {
...
+            while (StringUtils.isBlank(response2) && countAttempt <= attempt) {
+                countAttempt++;
+                response2 = liveImportClient.executeHttpGetRequest(1000, uriBuilder2.toString(), params2);
+            }
 
-            invocationBuilder = getRecordsTarget.request(MediaType.TEXT_PLAIN_TYPE);
-            response = invocationBuilder.get();
+            if (StringUtils.isBlank(response2)) {
+                throw new RuntimeException("After " + attempt
+                        + " attempts to contact the PabMed service, a correct answer could not be received."
```
Comment (tdonohue) [inline dspace-api/src/main/java/org/dspace/importer/external/pubmed/service/PubmedImportMetadataSourceServiceImpl.java:449]: PubMed is misspelled in this error as PabMed

File: dspace-api/src/main/java/org/dspace/importer/external/pubmed/service/PubmedImportMetadataSourceServiceImpl.java [hunk 3]
```diff
@@ -396,40 +407,57 @@ public FindMatchingRecords(Query q) {
...
+            while (StringUtils.isBlank(response) && countAttempt <= attempt) {
+                countAttempt++;
+                response = liveImportClient.executeHttpGetRequest(1000, uriBuilder.toString(), params);
+            }
 
-            String queryKey = getSingleElementValue(responseString, "QueryKey");
-            String webEnv = getSingleElementValue(responseString, "WebEnv");
+            if (StringUtils.isBlank(response)) {
+                throw new RuntimeException("After " + attempt
+                        + " attempts to contact the PabMed service, a correct answer could not be received."
```
Comment (tdonohue) [inline dspace-api/src/main/java/org/dspace/importer/external/pubmed/service/PubmedImportMetadataSourceServiceImpl.java:426]: PubMed is misspelled as PabMed

File: dspace-api/src/main/java/org/dspace/importer/external/pubmed/service/PubmedImportMetadataSourceServiceImpl.java [hunk 4]
```diff
@@ -226,24 +208,27 @@ public GetNbRecords(Query query) {
...
+            int countAttempt = 0;
+            while (StringUtils.isBlank(response) && countAttempt <= attempt) {
+                countAttempt++;
+                response = liveImportClient.executeHttpGetRequest(1000, uriBuilder.toString(), params);
+            }
 
-            String count = getSingleElementValue(responseString, "Count");
+            if (StringUtils.isBlank(response)) {
+                throw new RuntimeException("After " + attempt
+                        + " attempts to contact the PabMed service, a correct answer could not be received."
```
Comment (tdonohue) [inline dspace-api/src/main/java/org/dspace/importer/external/pubmed/service/PubmedImportMetadataSourceServiceImpl.java:224]: PubMed is misspelled as PabMed

**Review by tdonohue — COMMENTED** [review]
@Micheleboychuk : I haven't tested this yet, but gave the code a very quick review (I'll re-review later when testing it more thoroughly).  But, I immediately found a bunch of minor misspellings (you accidentally spelled it "PabMed" instead of "Pubmed" in a number of places, including configs).  So, we should clean up the misspellings when you have a chance. Thanks!

**Comment by Micheleboychuk** [comment]
@tdonohue  I have resolved the conflicts, it is now testable.

**Review by tdonohue — COMMENTED** [review]
Thanks @Micheleboychuk !  Overall, this works and it fixes the bug.  I just have a few minor suggestions to the error message inline below.

Beyond that, this fixes the bug, and the code looks good & passes all existing tests.

**Review by tdonohue — APPROVED** [review]
👍 Thanks @Micheleboychuk !  After that minor update to the logging, this all looks good to me.

**Pull request merged**